v2/internal/reflecthelpers/reflect_helpers.go

```diff
@@ -282,3 +282,85 @@ func getItemsField(listPtr client.ObjectList) (reflect.Value, error) {
 
 	return itemsField, nil
 }
+
+// SetProperty sets the property on the provided object to the provided value.
+// obj is the object to modify.
+// propertyPath is a dot-separated path to the property to set.
+// value is the value to set the property to.
+// Returns an error if any of the properties in the path do not exist, if the property is not settable,
+// or if the value provided is incompatible.
+func SetProperty(obj any, propertyPath string, value any) error {
+	if propertyPath == "" {
+		return errors.Errorf("property path was empty")
+	}
+
+	steps := strings.Split(propertyPath, ".")
+	return setPropertyCore(obj, steps, value)
+}
+
+func setPropertyCore(obj any, propertyPath []string, value any) (err error) {
+	// Get the underlying object we need to modify
+	subject := reflect.ValueOf(obj)
```

Comment:
do we need nil protection for `obj` here?

Reply:
Not here, it's safe when recursing. But we do need a check on the entry point, I'll add that.